examples/now.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std]
+/**
+ * This example shows a repeated timer combined with reading and displaying the current time in
+ * clock ticks.
+ **/
+use core::fmt::Write;
+use libtock::console::Console;
+use libtock::result::TockResult;
+use libtock::timer;
+use libtock::timer::Duration;
+
+#[libtock::main]
+async fn main() -> TockResult<()> {
+    const DELAY_MS: usize = 500;
+
+    let mut console = Console::new();
+    let mut previous_ticks = None;
+
+    for i in 0.. {
+        let mut timer_with_callback = timer::with_callback(|_, _| {});
+        let timer = timer_with_callback.init()?;
+        let current_clock = timer.get_current_clock()?;
+        let ticks = current_clock.num_ticks();
+        let frequency = timer.clock_frequency().hz();
+        writeln!(
+            console,
+            "[{}] Waited roughly {:?}. Now is {:?} = {:#010x} ticks ({:?} ticks since last time at {} Hz)",
+            i,
+            PrettyTime::from_ms(i * DELAY_MS),
+            PrettyTime::from_ms(current_clock.ms_f64() as usize),
+            ticks,
+            previous_ticks.map(|previous| ticks - previous),
+            frequency
+        )?;
+        previous_ticks = Some(ticks);
+        timer::sleep(Duration::from_ms(DELAY_MS as isize)).await?;
```

Comment:
This is unsafe in an non-obvious way: It actually overwrites the existing callback before it goes out of scope. We shouldn't recommend this and hopefully fix this soon by making the timer driver a singleton which can only be retrieved once.

Reply:
I was thinking that the scope could somehow end at the line where a variable isn't used anymore, but indeed that's not a safe assumption to make. Moved that to a separate function to avoid the issue.